src/Language/Haskell/GhclibParserEx/DynFlags.hs

```diff
@@ -20,6 +22,27 @@ import StringBuffer
 import HscTypes
 import GHC.LanguageExtensions.Type
 import Data.List
+import Data.Maybe
+import qualified Data.Map as Map
+
+-- | Parse a GHC extension
+readExtension :: String -> Maybe Extension
+readExtension = (`Map.lookup` exts)
+  where exts = Map.fromList [(show x, x) | x <- [minBound .. maxBound]]
+
+-- | Implicitly enabled/disabled extensions.
+extensionImplications :: [(Extension, ([Extension], [Extension]))]
+extensionImplications = map f $ Map.toList implicationsMap
+  where
+    f (e, ps) = (fromJust (readExtension e), ps)
+
+    implications :: [(String, Bool, Extension)]
+    implications = map f impliedXFlags
+      where f (e, flag, e') = (show e, flag == turnOn, e')
+
+    implicationsMap :: Map.Map String ([Extension], [Extension])
+    implicationsMap = Map.fromListWith f [(a, ([c | b], [c | not b])) | (a, b, c) <- implications]
```

Comment:
I'd probably "fuse" the impliciations into this function, e.g.

```suggestion
    implicationsMap = Map.fromListWith f [(show a, ([c | b], [c | not b])) | (a, bb, c) <- implications, let b = bb == turnOn]
```

Reply:
Ok!